app/src/main/java/br/org/cesar/knot_setup_app/view/scan/ScanPresenter.java

```diff
@@ -24,11 +27,58 @@
     private BluetoothWrapper bluetoothWrapper;
     private List<BluetoothDevice> deviceList;
     private UUID service;
+    private ScannerCallback scannerCallback;
+    private ScanCallback scanCallback;
 
     ScanPresenter(ViewModel viewModel, UUID service) {
         this.viewModel = viewModel;
         this.bluetoothWrapper = KnotSetupApplication.getBluetoothWrapper();
         this.service = service;
+
+        scannerCallback = new ScannerCallback() {
+            @Override
+            public void onScanComplete(android.bluetooth.le.ScanResult result) {
+                //Add to adapter list as a connecting option
+                //Check if current device was already inserted on list
+                for (BluetoothDevice device : deviceList) {
+                    if (device.getDevice().getAddress().equals(result.getDevice().getAddress())) {
+                        deviceList.remove(device);
+                        break;
+                    }
+                }
+                deviceList.add(new BluetoothDevice(result.getDevice(),result.getRssi()));
+                Collections.sort(deviceList);
+                viewModel.onDeviceFound(deviceList);
+            }
+
+            @Override
+            public void onScanFail() {
+                viewModel.onScanFail();
+            }
+        };
+
+        scanCallback  = new ScanCallback() {
+            @Override
+            public void onScanResult(int callbackType, ScanResult result) {
+                super.onScanResult(callbackType, result);
+                android.bluetooth.BluetoothDevice device = result.getDevice();
+                if(device.getName() != null) {
+                    LogWrapper.Log("NAME: " + device.getName(), Log.DEBUG);
+                    if(result.getScanRecord().getServiceUuids() != null) {
+                            for (ParcelUuid uuid :result.getScanRecord().getServiceUuids()) {
```

Comment:
I understand that you are just copying-pasting from BluetoothWrapper, but this for loop only adds complexity while being used for logs (or did I get something wrong) ?! 
If so, please remove.

Reply:
This Log is very important IMO, as it's constantly used for debugging. The for loop simply makes the log more readable. I also think that making a method to specifically print this Log is an overkill. What are your thoughts about this?

Reply:
You are inserting complexity only for debug purposes, I understand the use when debugging, but if this is used for debugging it should not affect production. 
Imagine when the app is running and you're not logging, you're just causing the application to iterate over `result.getScanRecord().getServiceUuids()` and do nothing.